src/Castle.Core/DynamicProxy/AbstractInvocation.cs

```diff
@@ -38,6 +42,15 @@ public abstract class AbstractInvocation : IInvocation
 			this.interceptors = interceptors;
 			this.proxiedMethod = proxiedMethod;
 			this.arguments = arguments;
+
+			var parameters = proxiedMethod.GetParameters();
+			if (ArgumentsUtil.IsAnyByRef(proxiedMethod.GetParameters()))
```

Comment:
It's occurred to me that it's probably not very clever to have this check executed upon every invocation. Perhaps the `AbstractInvocation` constructor should have an additional `bool proxiedMethodHasAnyByRefParameters` parameter that gets passed down from the custom-built invocation types (via `CompositionInvocation`); meaning the check would only run once per type generation, instead of once per invocation.